app/javascript/controllers/nav_controller.js

```diff
@@ -22,14 +22,17 @@ export default class extends Controller {
     //   when you tab in and out
     this.skipSandwichIcon = true;
 
-    document.addEventListener("click", (e) => {
-      // Must check both dropdowns otherwise you always close
-      // the dropdown because one or the other isn't being clicked
-      if (!this.dropdownTarget.contains(e.target) && !this.headerTarget.contains(e.target)) {
-        this.dropdownTarget.classList.remove('is-expanded');
-        this.collapseMobileNav();
-      }
-    });
+    document.addEventListener("click", this.clickOutsideDropdown.bind(this));
```

Comment:
Stimulus JS provides a first class API for binding to window events to remove the need for this particular pattern. It  makes the code way more simpler and intuitive to understand.

See:
* https://www.betterstimulus.com/events.html#register-global-events-in-your-markup
* https://stimulus.hotwired.dev/reference/actions#global-events

Reply:
Also I do see still 1 error in console on this branch.

![image](https://github.com/rubygems/rubygems.org/assets/193936/92872f3e-12b4-43fd-a06e-03c70238b3a9)

Reply:
Thanks for the help. I'm making these stimulus controllers better and I'll fix that error. Looks like a typo.

Reply:
@simi I can't figure out how you triggered that. I fixed it, but I thought I had done everything to test it so I'm stumped. I can't trigger the "focusin".

Reply:
I was able to trigger `focusin` on Chrome but not on Safari.